apps/src/applab/applab.js

```diff
@@ -510,21 +508,31 @@ Applab.getCode = function () {
 Applab.getHtml = function () {
   // This method is called on autosave. If we're about to autosave, let's update
   // levelHtml to include our current state.
-  if (Applab.isInDesignMode() && !Applab.isRunning() || Applab.levelHtml === '') {
+  if ($('#designModeViz').is(':visible')) {
     designMode.serializeToLevelHtml();
   }
   return Applab.levelHtml;
 };
 
+/**
+ * Sets Applab.levelHtml as well as #designModeViz contents.
+ * designModeViz is the source of truth for the app's HTML.
+ * levelHtml can be lazily updated from designModeViz via serializeToLevelHtml.
+ * @param html
+ */
 Applab.setLevelHtml = function (html) {
   if (html === '') {
-    if (window.dashboard) {
-      dashboard.project.clearHtml();
-    }
     Applab.levelHtml = '';
   } else {
     Applab.levelHtml = designMode.addScreenIfNecessary(html);
   }
+  var designModeViz = document.getElementById('designModeViz');
+  designMode.parseFromLevelHtml(designModeViz, true);
```

Comment:
An argument in favor or using options hashes for params, is that this would be easier to read as
```
designMode.parseFromLevelHtml(designModeViz, { allowDragging: true });
```
If we were in ES6 land, we could then even declare our function as follows (specifying that we have a hash with a field of allowDragging)
```
designMode.parseFromLevelHtml = function (element, { allowDragging }) {
  // allowDragging is now a local variable
}
```
In any case, no changes are necessary for this PR :)